VERSIONING_POLICY.md

```diff
@@ -0,0 +1,33 @@
+# Open OnDemand Versioning Policy
+## Version Numbering
+Open OnDemand code releases utilize a XXX.YYY.ZZZ numbering scheme (e.g. 2.0.29), where:
+* XXX is the major version number
+* YYY is the minor version number
+* ZZZ is the patch version number
+
+## Major Versions
+New major version numbers will typically be implemented for any of the following reasons:
+* Significant breaking changes that prevent seamless upgrading from previous versions
+* Significant deprecations related to software dependencies
+* Large sets of new features or functionalities
+
+## Minor Versions
+New minor version numbers will typically be implemented for any of the following reasons:
+* New features of functionalities
+* Large sets of bug fixes
+
+## Patch Versions
+New patch version numbers will typically be implemented for any of the following reasons:
+* Security updates
+* Minor bug fixes
```

Comment:
The Changleog talks about semantic versioning, though IDK if we're following it anymore because it's centered more around compatibility than features.

https://github.com/OSC/ondemand/blob/a448400a6b1a47b92e2fe8322abf26530ff63d77/CHANGELOG.md?plain=1#L6

Reply:
Do you mean the motivation of using it to help with managing dependencies? I can see what you mean, but it seems like adding functionality is a cause of dependency introduction and change, so it is still coupled (maybe loosely) to that notion of managing dependencies. I don't think anyone will go and check anyway to ensure we are following semantic versioning to the T anyway, most people in the space are vaguely familiar with the idea of it and following another scheme would be confusing it seems like. What were you thinking?

Reply:
Feel free to make changes.  I wasn't aware of the semantic versioning page and just reviewed it.  It seems like we are generally abiding by the spirit of that, although some of the details don't seem to translate too well (e.g. all the references to APIs and backward compatibly don't make much sense in the OOD world to me).  Maybe we should add a section that links to the semantic page and says we are trying to follow the spirit of it.  

I don't think we should just link to the semantic page and eliminate all the content I propose because that page is far too detailed and it's easy to 'miss the forest for the trees'.

Reply:
Just realized I should maybe mention something else related to all this.  I think there are 2 very distinct audiences for this policy, and I suspect many of the comments so far are being skewed heavily towards one or the other.  In mind mind those audiences are:
1. Technical code contributors such as the OSC software developers and community contributors such as Harvard and CSC-Finland.
2. General OOD community members, who often are in an administrative or client support role at their institution.  

I suspect people in group 1 are more interested in the specific nuts and bolts stuff within this policy (i.e. much of the stuff like whats on the semantic versioning page).  But people in group 2 want the high level stuff like general timelines and what the expected support periods are.

If there is a better way to convey information to both those groups via different documents / communications channels I'm open to hearing suggestions.